### PR TITLE
Tidy up creation of SierraTransformable

### PR DESCRIPTION
**Pull request by alexwlchan**
Follows #2444, spun out of #2432.

### CODE REVIEW

File: catalogue_pipeline/transformer/src/test/scala/uk/ac/wellcome/platform/transformer/transformers/SierraTransformableTransformerTest.scala [hunk 1]
```diff
@@ -18,14 +18,11 @@ class SierraTransformableTransformerTest
...
-      sourceId = id,
-      maybeBibData = Some(createSierraBibRecordWith(id = id)),
-      itemData = Map(
-        "5151515" -> createSierraItemRecordWith(id = "5151515"),
-        "5252525" -> createSierraItemRecordWith(id = "5252525")
+    val sierraTransformable = createSierraTransformableWith(
+      maybeBibRecord = Some(createSierraBibRecordWith(id = id)),
+      itemRecords = List(
+        createSierraItemRecordWith(id = "5151515"),
+        createSierraItemRecordWith(id = "5252525")
```
Comment (mklander) [inline catalogue_pipeline/transformer/src/test/scala/uk/ac/wellcome/platform/transformer/transformers/SierraTransformableTransformerTest.scala]: 👏

File: catalogue_pipeline/transformer/src/test/scala/uk/ac/wellcome/platform/transformer/transformers/sierra/SierraItemsTest.scala [hunk 2]
```diff
@@ -16,41 +15,31 @@ class SierraItemsTest extends FunSpec with Matchers with SierraDataUtil {
 
   describe("extractItemData") {
     it("parses instances of SierraItemData") {
-      val item1 = createSierraItemData
-      val item2 = createSierraItemData
+      val itemData = (1 to 2).map { _ => createSierraItemData }
+      val itemRecords = itemData.map { createSierraItemRecordWith }.toList
```
Comment (mklander) [inline catalogue_pipeline/transformer/src/test/scala/uk/ac/wellcome/platform/transformer/transformers/sierra/SierraItemsTest.scala]: `createSierraItemRecordWith` looks a bit odd here is it with defaults, `createSierraItemRecord`?

Comment (alexwlchan) [reply catalogue_pipeline/transformer/src/test/scala/uk/ac/wellcome/platform/transformer/transformers/sierra/SierraItemsTest.scala]: No, we’re passing in the instances of SierraItemData. I’ll tweak to clarify.

**Comment by alexwlchan** [comment]
Oops, I mixed in some SierraRecord stuff onto this branch. 🙈

**Pull request merged**